src/librustc_trans/type_.rs

```diff
@@ -299,6 +300,27 @@ impl Type {
             llvm::LLVMGetIntTypeWidth(self.to_ref()) as u64
         }
     }
+
+    pub fn from_integer(cx: &CrateContext, i: layout::Integer)->Type {
```

Comment:
I don't think so.  @eddyb says that at the LLVM level, the distinction isn't made.  You can confirm this by observing that `int_from_ty` and `uint_from_ty` do the same thing.